File: src/main/java/ac/dnd/server/admission/infrastructure/web/mapper/ApplicantWebMapper.java
package ac.dnd.server.admission.infrastructure.web.mapper;

import org.springframework.stereotype.Component;

import ac.dnd.server.admission.application.dto.ApplicantStatusCheckCommand;
import ac.dnd.server.admission.infrastructure.web.dto.request.ApplicantStatusCheckRequest;
import lombok.RequiredArgsConstructor;

@Component
@RequiredArgsConstructor
public class ApplicantWebMapper {

	public ApplicantStatusCheckCommand statusCheckRequestToCommand(
		ApplicantStatusCheckRequest request,
		final Long eventId
	) {
		return new ApplicantStatusCheckCommand(
			eventId,
			request.name(),
			request.email()

		);
	}

}

File: src/main/kotlin/ac/dnd/server/admission/infrastructure/web/mapper/ApplicantWebMapper.kt
package ac.dnd.server.admission.infrastructure.web.mapper

import ac.dnd.server.admission.application.dto.ApplicantStatusCheckCommand
import ac.dnd.server.admission.infrastructure.web.dto.request.ApplicantStatusCheckRequest
import org.springframework.stereotype.Component

@Component
class ApplicantWebMapper {

    fun statusCheckRequestToCommand(
        request: ApplicantStatusCheckRequest,
        eventId: Long
    ): ApplicantStatusCheckCommand {
        return ApplicantStatusCheckCommand(
            eventId,
            request.name,
            request.email
        )
    }
}
